prime_factor/prime_factor.c

```diff
@@ -0,0 +1,37 @@
+#include <stdio.h>
+
+void generate_primes(int *ar, int *cnt) {
+    *cnt = 0;
+    for (int i = 2; i < 60000 && *cnt < 1000; ++i) {
```

Comment:
So they are there to prevent overflow and 1000 is the array size which is already predefined. At large i can pass them as arguments to function.